### PR TITLE
Move types to nmr processing

### PR DESCRIPTION
**Pull request by jobo322**
use types from nmr-processing

### CODE REVIEW

**Comment by targos** [comment]
@lpatiny Doesn't it make more sense to move the types to `cheminfo-types` ?

File: .eslintrc.yml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 extends: cheminfo-typescript
 rules:
+  '@typescript-eslint/naming-convention': off
```
Comment (targos) [inline .eslintrc.yml:3]: Why do you disable this rule?

File: src/types/WorkSpace.ts [hunk 2]
```diff
@@ -0,0 +1,242 @@
+import { FifoLoggerOptions } from 'fifo-logger';
+import type { SplitPaneSize } from 'react-science/ui';
```
Comment (targos) [inline src/types/WorkSpace.ts:2]: We usually don't do that, but depending on the big `react-science` library just for this type is overkill. I think it's ok here to copy the type and remove the dependency.

```ts
type SplitPaneSize = `${number}%` | `${number}px`;
```

File: .eslintrc.yml (hunk 1 above)
Comment (jobo322) [reply .eslintrc.yml:3]: In nmrium the constant are in upper case. In nmrium this rule is disable that is why I do the same here. But seems it is not needed any more. Thanks

Comment (targos) [reply .eslintrc.yml:3]: The change is still here.

Comment (jobo322) [reply .eslintrc.yml:3]: yes, it is true, I push the changes in another branch. let me rebase it

File: src/read/__tests__/read.test.ts [hunk 3]
```diff
@@ -263,8 +263,8 @@ describe('filtering a jcamp with assignment', () => {
   });
 });
 
-describe('varian failing test case', () => {
-  it('processed FID should has the same scale that jcamp data', async () => {
+describe.only('varian failing test case', () => {
+  it.only('processed FID should has the same scale that jcamp data', async () => {
```
Comment (targos) [inline src/read/__tests__/read.test.ts:267]: ⚠️

File: src/types/Options/AnalysisOptions.ts [hunk 4]
```diff
@@ -1,19 +1,19 @@
...
+  FORMULA: 'FORMULA',
+};
+export const AnalisisColumnsValuesKeys = {
+  RELATIVE: 'relative',
+  ABSOLUTE: 'absolute',
+  MIN: 'min',
+  MAX: 'max',
+};
+export type ColumnsTypes = 'NORMAL' | 'FORMULA';
+export type ColumnsValuesKeys = 'relative' | 'absolute' | 'min' | 'max';
```
Comment (targos) [inline src/types/Options/AnalysisOptions.ts:12]: You should compute these types, and please use the same name as the objects:

```
export type AnalysisColumnsTypes = (typeof AnalysisColumnsTypes)[keyof typeof AnalysisColumnsTypes];
```

File: src/utilities/getNucleusFromMetadata.ts [hunk 5]
```diff
@@ -58,16 +59,16 @@ function getNucleusFrom2DExperiment(experiment: string) {
     return [];
   }
   experiment = experiment.toLowerCase();
-  if (experiment.includes('jres')) {
+  if (experiment.match('jres')) {
```
[the remark this answers is not in the document]
Comment (jobo322) [reply src/utilities/getNucleusFromMetadata.ts:71]: not all the experiments are called jres, there are several variants

File: src/types/Options/AnalysisOptions.ts (hunk 4 above)
Comment (jobo322) [reply src/types/Options/AnalysisOptions.ts:12]: ok, thanks

File: src/read/__tests__/read.test.ts (hunk 3 above)
Comment (jobo322) [reply src/read/__tests__/read.test.ts:267]: it is done in the last push

File: src/utilities/getNucleusFromMetadata.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (jobo322) [reply src/utilities/getNucleusFromMetadata.ts:71]: sorry, it is wrong sorry, it is not a array it is my fault

**Review by targos — APPROVED** [review]

**Pull request merged**